Some changes

mertainer opened the pull request:
Don't have much experience with C# please comment

+ 'No Fight' to Nuis;
+ Some error messages to AH;
+ Total Honor / PvP kills to characters (missing call);
+ Required grade / main grade to crafts;
+ Fix portals not despawning / other portal not dying if one is killed.
+ Fix portals attacking player
+ Fix other players disappearing when mounted
+ Fix mount despawn bug
+ Fix unsealing items not keeping grade
+ Fix crafted items not keeping grade
+ Added grade upgrade to weapon crafting
+ Fix bug when logging in while other players are mounted
+ Added /savespawn command to write extra spawns to separate json

AAGene commented on AAEmu.Game/Core/Managers/UnitManagers/NpcManager.cs:
```diff
@@ -122,6 +122,10 @@ public Npc Create(uint objectId, uint id)
 
             npc.Hp = npc.MaxHp;
             npc.Mp = npc.MaxMp;
+
+            if (template.Id == 7989 || template.Id == 7988 || template.Id == 7987) // If is a nui, better way to do this?
+                NuiManager.Instance.Add(npc);
+
```
There is a better way indeed.

Nuis (the statues) are Flytrap AIs with a OnSpawn skill that creates a doodad with a DoodadFuncClout that handles the creation of the area trigger.

So you'd be better off by using the OnSpawn skills from np_skills when the NPC spawns.

mertainer replied:
Looked up the SQLite DB and now I see what you mean. 

This onSpawn skill seems to be 'skill_use_condition_id' == 5 on 'np_skills'? Since it spawns the 'Peace Zone' doodad (3388) from an 'InteractionEffect' I'm assuming this is it.
Flytrap would be the AI for casting the bubble on players hitting each other, unsure how it checks if players hitting each other. Maybe 'aggro_link_special_rule_id' from 'npcs'? 

For this would have to implement 'np_skills' and 'npc_ai_params'.

mertainer replied:
Removed this implementation

AAGene commented on AAEmu.Game/Core/Packets/G2C/SCRankCharacterPacket.cs:
```diff
@@ -0,0 +1,24 @@
+
+        public override PacketStream Write(PacketStream stream)
+        {
+            // "Fixes" infinite loading, sends less causes packet error
+            stream.Write(0); 
+            stream.Write(0);
+            stream.Write(0);
+            stream.Write(0);
+            stream.Write(0);
+            return stream;
```
Instead of sending an empty packet, you should be sending the structure, this change is not useful as it is only a hackfix, and not a proper fix

AAGene commented on AAEmu.Game/Models/Game/Units/Mount.cs:
```diff
@@ -392,5 +414,25 @@ public override void BroadcastPacket(GamePacket packet, bool self)
             }
         }
+
+        private void updatePassengers()
```
Naming convetion:

UpdatePassengers()

AAGene commented on AAEmu.Game/Models/Game/Units/Mount.cs:
```diff
@@ -392,5 +414,25 @@ public override void BroadcastPacket(GamePacket packet, bool self)
             }
         }
+
+        private void updatePassengers()
+        {
+            foreach (var passenger in getPassengers())
+            {
+                passenger.SetPosition(Position);
+            }
+        }
+
+        public List<Character> getPassengers()
```
Naming convention: GetPassengers()

AAGene commented on AAEmu.Game/Core/Managers/CraftManager.cs:
```diff
@@ -114,10 +115,10 @@ public void Load()
                         }
                     }
-                }
+                }*/
```
Why would you comment this?

mertainer replied:
From what I've tested that table seems to do nothing. In GainItemPack it was causing regular items to be flagged as a trade pack which wouldn't allow more than 1x craft at a time

AAGene commented on AAEmu.Game/Core/Managers/AuctionManager.cs:
```diff
@@ -388,7 +403,7 @@ public AuctionItem CreateAuctionItem(Character player, Item itemToList, int star
                 EndTime = DateTime.UtcNow.AddHours(timeLeft),
                 LifespanMins = 0,
                 Type1 = 0,
-                WorldId = 0,
+                WorldId = 1,
```
WorldID needs to be 0 for the game to properly assume that the world is the main world.

mertainer replied:
Shouldn't the WorldId be 1? I think this is for multi-server stuff.

This fixed seeing your own bids in AH as other people's bids, which allowed you to re-bid on the same item

AAGene replied:
I see. Could you rename it to GameServer ID then ?

mertainer commented:
Removed Nui code since it's better to implement after Npc AI is moved here.
Added working recalls - uses portals JSON. TODO: Needs updating for missing areas.

ZeromusXYZ commented:
I've added some very basic code so kills actually give and subtract honor for testing
